pybossa/api/__init__.py

```diff
@@ -461,3 +463,63 @@ def task_gold(project_id=None):
     task_repo.update(task)
 
     return Response(json.dumps({'success': True}), 200, mimetype="application/json")
+
+
+@jsonpify
+@login_required
+@csrf.exempt
+@blueprint.route('/task/<task_id>/services', methods=['POST'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_service_request(task_id):
+    """Proxy service call"""
+    proxy_service_config = current_app.config.get('PROXY_SERVICE_CONFIG', None)
+    task = task_repo.get_task(task_id)
+    payload = request.json
+
+    if not task or not proxy_service_config:
+        return abort(400)
+
+    scheduler, timeout = get_project_scheduler_and_timeout(
+        task.project_id)
+
+    task_locked_by_user = False
+
+    if scheduler in (Schedulers.locked, Schedulers.user_pref):
+        task_locked_by_user = has_lock(task.id, current_user.id, timeout)
+
+    if task_locked_by_user:
+        service = _get_valid_service(task_id, payload, proxy_service_config) if isinstance(payload, dict) else None
+        if service:
+            url = '{}/{}/1/{}'.format(proxy_service_config['uri'], service['name'], payload['service_version'])
+            headers = service['headers']
+            ret = requests.post(url, headers=headers, json=payload['data'])
+            return Response(json.dumps(ret.json()), 200, mimetype="application/json")
+
+    current_app.logger.info(
+        'Task id {} with lock-status {} by user {} with this payload {} failed.'
+        .format(task_id, task_locked_by_user, current_user.id, payload))
+    return abort(403)
+
+
+def _get_valid_service(task_id, payload, proxy_service_config):
+    whitelist_regex = regex.compile(ur"[^a-zAz0-9\.\u00BC-\u00BE\u2150-\u215E\s]")
+    service_name = payload.get('service_name', None)
+    service_version = payload.get('service_version', None)
+    service_data = payload.get('data', None)
+    service_request = service_data.keys()[0] if isinstance(service_data, dict) and len(service_data.keys()) == 1 else None
+
+    if service_name == proxy_service_config['services']['autocomplete_service']['name']:
```

Comment:
as per comment [here](https://github.com/bloomberg/pybossa/pull/309/files#diff-e23f749f9cbf6993dd73b23a99f4707dR471), service name can be obtained from url passed as input